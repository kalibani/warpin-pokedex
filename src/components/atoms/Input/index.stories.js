// Stories Input Component
// --------------------------------------------------------
import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';
import Input from './index';
import Notes from './readme.md';

storiesOf('Atoms', module).add('Input', () => (
  <Input />
), {
  info: { inline: true, header: false, text: Notes }
});
