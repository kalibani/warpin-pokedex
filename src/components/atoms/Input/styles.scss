/* ==========================================================================
Styles for Input component
========================================================================== */

// import mixin and variables
@import "../../../assets/styles/variables";

// Start styles Input component
.a-input {

}
