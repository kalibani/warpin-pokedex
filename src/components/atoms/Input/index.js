// Input Component
// --------------------------------------------------------

import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

const Input = ({ propsName }) => (
  <tag>
    Functional Component (Stateless Component)
  </tag>
);

Input.propTypes = {
  propsName: PropTypes.string
};

Input.defaultProps = {
  propsName: ''
};

export default Input;
