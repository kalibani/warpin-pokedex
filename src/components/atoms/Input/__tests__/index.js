// Test Input Component
// --------------------------------------------------------

import React from 'react';
import Input from '../index';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('<Input />', () => {
  describe('onClick()', () => {
    test('successfully calls the onChange handler and match the snapshot', () => {
      const mockOnChange = jest.fn();
      const wrapper = shallow(
        <Input onChange={mockOnChange}/>
      );

      wrapper.find('input').simulate('change');
      expect(mockOnChange.mock.calls.length).toEqual(1);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});


