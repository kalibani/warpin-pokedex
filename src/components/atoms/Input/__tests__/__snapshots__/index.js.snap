// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> onClick() successfully calls the onChange handler and match the snapshot 1`] = `
<Fragment>
  <input
    autoComplete="on"
    className=""
    disabled={false}
    id=""
    maxLength=""
    name=""
    onBlur={[Function]}
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder=""
    tabIndex="0"
    type="text"
    value=""
  />
</Fragment>
`;
